bundle/filefs.go

```diff
@@ -0,0 +1,73 @@
+// +build go1.16
+
+package bundle
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type dirLoaderFS struct {
+	filesystem fs.FS
+	files      []string
+	idx        int
+}
+
+// Interface guard for dirLoaderFS to validate that it implements the DirectoryLoader interface
+var _ DirectoryLoader = (*dirLoader)(nil)
+
+// NewFSLoader returns a basic DirectoryLoader implementation
+// that will load files from a fs.FS interface
+func NewFSLoader(filesystem fs.FS) DirectoryLoader {
+	d := dirLoaderFS{
+		filesystem: filesystem,
+	}
+
+	return &d
+}
+
+// NextFile iterates to the next file in the directory tree
+// and returns a file Descriptor for the file.
+func (d *dirLoaderFS) NextFile() (*Descriptor, error) {
+	// build a list of all files we will iterate over and read, but only one time
+	if d.files == nil {
+		d.files = []string{}
```

Comment:
Appending to a nil slice is fine, so you could drop the initialization (`d.files = []string{}`) (👉 https://play.golang.org/p/yxp2ahK2jCl)